From 2caa6ee1c2fbb020159413e456e5f4ac75685ed9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.epitre@orange.fr>
Date: Sat, 10 Nov 2018 17:15:33 +0100
Subject: [PATCH] anaconda: add skip_grub parameter and allow boot encryption
 and the lvmlv format
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On coreboot systems, as grub2 can be used a payload it is possible to have all the partitions encrypted.

Based on commits bf803e7 and 62cb1ca.

Signed-off-by: Frédéric Pierret (fepitre) <frederic.epitre@orange.fr>
---
 pyanaconda/bootloader.py | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/pyanaconda/bootloader.py b/pyanaconda/bootloader.py
index 672aa66ec..3f336382a 100644
--- a/pyanaconda/bootloader.py
+++ b/pyanaconda/bootloader.py
@@ -1448,12 +1448,13 @@ class GRUB2(GRUB):
     # XXX we probably need special handling for raid stage1 w/ gpt disklabel
     #     since it's unlikely there'll be a bios boot partition on each disk
 
-    @property
-    def stage2_format_types(self):
-        if productName.startswith("Red Hat "):              # pylint: disable=no-member
-            return ["xfs", "ext4", "ext3", "ext2"]
-        else:
-            return ["ext4", "ext3", "ext2", "btrfs", "xfs"]
+    stage2_format_types = ["ext4", "ext3", "ext2", "btrfs", "xfs"]
+
+    def __init__(self):
+        super().__init__()
+        self.encryption_support = True
+        self.stage2_format_types += ["lvmlv"]
+        self.skip_bootloader = flags.cmdline.getbool("skip_grub", False)
 
     #
     # grub-related conveniences
-- 
2.19.2

